### PR TITLE
Click event improvements

### PR DESCRIPTION
**Pull request by jbeezley**
This is motivated by a conversation with @mbertrand about the click events.  With this PR, click events will always prevent other synthetic events (pan, zoom, brush, etc.) from occurring from the same native event.  Exactly how a click is detected is configurable via the options:
 * `options.click.enabled`: to enable globally
 * `options.click.buttons`: to enable per mouse button
 * `options.click.duration`: trigger a click only when the time between `mousedown` -> `mouseup` is less than this many milliseconds.  This check is disabled when the value is zero (the default).
 * `options.click.cancelOnMove`: a click will not be triggered if there is a `mousemove` between the `mousedown` and `mouseup`

This will almost certainly conflict trivially with #455.  I'll rebase as necessary.

### CODE REVIEW

File: src/core/mapInteractor.js [hunk 1]
```diff
@@ -779,10 +826,34 @@ geo.mapInteractor = function (args) {
...
+
     m_this._getMouseButton(evt);
     m_this._getMouseModifiers(evt);
 
-    // fire a click event here
+    // cancel any ongoing pan action
+    m_this.cancel('pan');
+
+    // unbind temporary handlers on document
+    $(document).off('.geojs');
```
Comment (aashish24) [inline src/core/mapInteractor.js]: @jbeezley why this is required?

**Comment by aashish24** [comment]
@jbeezley @mbertrand  I liked the idea +1. It would be nice if we can add some documentation for in in the RTD.

File: src/core/mapInteractor.js (hunk 1 above)
Comment (jbeezley) [inline src/core/mapInteractor.js]: `mousedown` adds event handlers to `document` to track the mouse even if it moves outside the map.  This is only needed during an ongoing pan/zoom so the handler is added temporary.  This just ensures that the handlers get unbound when stopping the pan action that had been initiated.

Comment (aashish24) [inline src/core/mapInteractor.js]: Ah, I see, thanks.

**Comment by aashish24** [comment]
@jbeezley didn't run the code, but in general it LGTM :+1:

**Comment by aashish24** [comment]
LGTM :+1:

**Pull request merged**